test/parallel_api/numeric/numeric.ops/scan.pass.cpp

```diff
@@ -230,6 +231,13 @@ test_matrix(Out init, BinaryOp binary_op, Out trash)
                                     out.end(), expected.begin(), expected.end(), in.size(), init, binary_op, trash);
 #endif
 #endif
+        };
+
+        //perform regular a scan algorithm
+        __scan_invoker(out);
+
+        //perform an in-place scan algorithm
+        __scan_invoker(in);
```

Comment:
I don't see any changes to restore the values of the input sequence between tests. Since `in` sequence is modified for the in-place scan, we run the individual tests on different values, no? I don't think this is what we want, or am I missing something?